### PR TITLE
feat: filter logs for only the block being queried

### LINKED ISSUE #5006: [RPC] Flaky `Filecoin.EthGetBlockReceipts` test
## Describe the bug

CI failure: https://github.com/ChainSafe/forest/actions/runs/12031124974/job/33540320587

and https://github.com/ChainSafe/forest/actions/runs/12031651546/job/33542048826?pr=4959

## To reproduce

1. Download latest calibnet snapshot
2. Run forest-tool api compare with `-n 800`
3. See error in comments of this issue or in the CI links above

## Log output

<details>
  <summary>Log Output</summary>
  
  ```Paste log output here
  paste log output...
  ```
</details>

## Environment (please complete the following information):

- OS:
- Branch/commit
- Hardware

### PR DESCRIPTION
**Pull request by virajbhartiya**
## Summary of changes

Changes introduced in this pull request:
[`src/rpc/methods/eth.rs`](diffhunk://#diff-4edd6390405df0ec5b2500428ceb98f2ccd3ff7889c95f27e692100726a66e1aR1157-R1167): Added logic to filter events based on the current block hash in the `new_eth_tx_receipt` function. This ensures that only events matching the current block hash are retained.

## Reference issue to close (if applicable)
Closes #5006

## Change checklist

- [ ] I have performed a self-review of my own code,
- [ ] I have made corresponding changes to the documentation. All new code adheres to the team's [documentation standards](https://github.com/ChainSafe/forest/wiki/Documentation-practices),
- [ ] I have added tests that prove my fix is effective or that my feature works (if possible),
- [ ] I have made sure the [CHANGELOG][1] is up-to-date. All user-facing changes should be reflected in this document.

[1]: https://github.com/ChainSafe/forest/blob/main/CHANGELOG.md

### CODE REVIEW

File: src/rpc/methods/eth.rs [hunk 1]
```diff
@@ -1154,6 +1154,17 @@ async fn new_eth_tx_receipt<DB: Blockstore + Send + Sync + 'static>(
...
+
+    let current_block_hash = receipt.block_hash.clone();
+    events.retain(|event| {
+        if let Ok(block_hash) = event.tipset_key.cid() {
+            let event_block_hash: EthHash = block_hash.into();
+            event_block_hash == current_block_hash
+        } else {
+            false
+        }
+    });
```
Comment (hanabi1224) [inline src/rpc/methods/eth.rs:1166]:
```suggestion
    let current_block_hash = &receipt.block_hash;
    events.retain(|event| {
        if let Ok(block_hash) = event.tipset_key.cid() {
            let event_block_hash: EthHash = block_hash.into();
            &event_block_hash == current_block_hash
        } else {
            false
        }
    });
```

File: src/rpc/methods/eth.rs [hunk 2]
```diff
@@ -1154,6 +1154,17 @@ async fn new_eth_tx_receipt<DB: Blockstore + Send + Sync + 'static>(
 
     let mut events = vec![];
     EthEventHandler::collect_events(ctx, &parent_ts, None, &mut events).await?;
+
+    let current_block_hash = &receipt.block_hash;
+    events.retain(|event| {
+        if let Ok(block_hash) = event.tipset_key.cid() {
+            let event_block_hash: EthHash = block_hash.into();
+            event_block_hash == *current_block_hash
```
Comment (hanabi1224) [inline src/rpc/methods/eth.rs:1162]: Why copy `current_block_hash` in the loop instead of comparing the borrow?

Comment (virajbhartiya) [reply src/rpc/methods/eth.rs]: Oh yes, I should've reference it

**Comment by elmattic** [comment]
Could you create some `EthFilterSpec` with the current block hash, and reuse filtering logic of `::collect_events`?

**Comment by elmattic** [comment]
Let's close this PR if we're not 100% how to fix this flakyness.

**Pull request closed without merging**